certbot-ci/certbot_integration_tests/nginx_tests/test_main.py

```diff
@@ -13,5 +17,41 @@ def context(request):
         integration_test_context.cleanup()
 
 
-def test_hello(context):
-    print(context.directory_url)
+@pytest.mark.parametrize('certname_pattern, params, context', [
+    ('nginx.{0}.wtf', ['run'], True),  # Passing True here makes the context
+                                       # fixture start Nginx with a default server.
+    ('nginx2.{0}.wtf', ['--preferred-challenges', 'http'], True),
+    # Overlapping location block and server-block-level return 301
+    ('nginx3.{0}.wtf', ['--preferred-challenges', 'http'], True),
+    # No matching server block; default_server exists
+    ('nginx4.{0}.wtf', ['--preferred-challenges', 'http'], True),
+    # No default server in Nginx starting to this point.
+    ('nginx5.{0}.wtf', ['--preferred-challenges', 'http'], False),
```

Comment:
Can we add `# No matching server block; default_server does not exist` above this?

Reply:
Done.